plugins/syntax/numeral.ml

```diff
@@ -109,6 +108,11 @@ let vernac_numeral_notation local ty f g scope opts =
   in
   let opt r = app (mkRefC (q_option ())) r in
   let constructors = get_constructors tyc in
+  (* Before, this was taking the state from the proof, I do believe
+     this was wrong *)
+  let env = Global.env () in
```

Comment:
wrong env

Reply:
Why is the wrong env? I think before it was wrong.

Reply:
I could not figure out what `f` potentially is.
If you there is a code path leading here where `f` is a context variable (parsed/globalized as such) then you'd get a not_found rather than a more polite error (such as f has not the expected type).

Reply:
Well the point here is that if such a term is build by this plugin it would fail on registration time too, so let's state it this way, the plugin doesn't support using non-global envs.

Some bold soul should really make a different type for the proof-time env....